apis/python/src/tiledbsoma/tiledb_object.py

```diff
@@ -124,134 +115,41 @@ def exists(self) -> bool:
 
         [lifecycle: experimental]
         """
+        # We already opened this, so it should always exist.
+        return not self._was_deleted
```

Comment:
to confirm: the API is not supporting the edge case where the underlying data was deleted after open. 

I consider this acceptable, but the old code would catch that case, so there is a change in semantics.

Reply:
Second thought:  why isn't this a class method, which takes a URI?  The idea being that create/delete/exists all should be class methods, outside of the lifecycle of the actual instance.

Reply:
I've removed this method, since as you note we're not supporting the weird "pull the rug out" edge case. It could be added back in as a classmethod, but I feel like most uses of an "exists" function would be in the form of "does it exist, if so open, else create", which is better handled by the easier-to-ask-forgiveness model of "try and open, otherwise create".

HOWEVER!

I do think there is room for a unified "open-or-create" function which does both of these at the same time since that's a common use case (compare: opening a file in `r+` mode), and it could save a lot of user duplication in that respect.

Reply:
I agree.  I also think there is room for an "open URI and return it" without knowing the type a priori.

A la `tiledbsoma.open(uri)`